### PR TITLE
LibJS: Add the WeakMap built-in object

### PR DESCRIPTION
**Pull request by IdanHo**
This fixes 84 test262 test cases.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/WeakContainer.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibJS/Heap/Heap.h>
+
+namespace JS {
+
+class Heap;
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/WeakContainer.h:13]: We already include `Heap.h`, why is this needed?

File: Userland/Libraries/LibJS/Runtime/WeakMap.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#include <AK/HashMap.h>
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/Object.h>
+#include <LibJS/Runtime/WeakContainer.h>
+
+namespace JS {
+
+class WeakMap : public Object
+    , public WeakContainer {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/WeakMap.h:17]: The preferred formatting is

```suggestion
class WeakMap
    : public Object
    , public WeakContainer {
```

Also, please make this `final`.

File: Userland/Libraries/LibJS/Runtime/WeakSet.h [hunk 3]
```diff
@@ -9,10 +9,12 @@
 #include <AK/HashTable.h>
 #include <LibJS/Runtime/GlobalObject.h>
 #include <LibJS/Runtime/Object.h>
+#include <LibJS/Runtime/WeakContainer.h>
 
 namespace JS {
 
-class WeakSet : public Object {
+class WeakSet : public Object
+    , public WeakContainer {
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/WeakSet.h:17]:
```suggestion
class WeakSet final
    : public Object
    , public WeakContainer {
```

File: Userland/Libraries/LibJS/Runtime/WeakContainer.h (hunk 1 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/WeakContainer.h]: this was left over from testing, removed.

File: Userland/Libraries/LibJS/Runtime/WeakMap.h (hunk 2 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/WeakMap.h]: fixed

File: Userland/Libraries/LibJS/Runtime/WeakSet.h (hunk 3 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/WeakSet.h]: fixed

**Comment by IdanHo** [comment]
FYI, looks like the ALL_DEBUG build failed due to dependency mirror issues

**Comment by linusg** [comment]
Already saw it fully green earlier, it's all good :)
